app/helpers/ledger.js

```diff
@@ -0,0 +1,163 @@
+import { wallet } from "wallet-preload-shim";
+import { strHashToRaw } from "helpers";
+import { default as blake } from "blake-hash";
+import * as bs58 from "bs58";
+import toBuffer from "typedarray-to-buffer";
+import { getTxFromInputs } from "../actions/TransactionActions";
+
+export function addressPath(branch, index) {
+  const prefix = "44'/42'/0'/";
+  const i = (index || 0).toString();
+  const b = (branch || 0).toString();
+  return prefix + b + "/" + i;
+}
+
+// fixPubKeyChecksum replaces the sha256 checksum, or last four bytes, of a
+// pubkey with a blake256 checksum.
+export function fixPubKeyChecksum(pubKey) {
+  const buff = bs58.decode(pubKey).slice(0, -4);
```

Comment:
This will make it into a valid xpub, ok.

**However**, this means you can't go from seed to wallet using the standard decred software or my BIP0039 -> raw seed tool.

My concern here is that, if the user (say) gets its ledger stolen and wants to withdraw all funds as fast as possible using only software, there will be no fully compatible software to do so, because this particular path from mnemonic seed -> raw seed bytes -> account is not replicated on any software.

At the very least we'll need a tool that can go from the ledger mnemonic seed to either a raw seed or account xprivs that can be imported into decrediton/dcrwallet so that we can accurately recreate the wallet in software only if needed.

Reply:
> However, this means you can't go from seed to wallet using the standard decred software or my BIP0039 -> raw seed tool.

Where was the tool again? In the decred repo? I'm unsure why it would not work but I trust you are correct.

Reply:
@matheusd I was able to use the tool at https://github.com/matheusd/bip39-to-dcr-seed to restore a decred wallet from seed.

I made a mainnet wallet with decrediton and ledger. Send it a small amount of funds. After restoring using your tool, the seed words, and dcrwallet I was able to access the funds. It works!

Reply:
I should mention that the tool will not work on testnet because the ledger testnet app forces cointype 42' although dcrwallet wants 1'. This is fine for mainnet though.

Reply:
Does that testnet cointype restriction apply even if you change the following line:

https://github.com/decred/decrediton/compare/632d45593fe742186609bffbc1b35c3ed2e59580..77583e96a8774c61c8ee2a3c4891216e93b24410#diff-bebc2af593a6d40794e960263162e817fe54e22c9e72586a4faf7e766ad3d71cR194

?

Reply:
Change "44'/42'/0'" to "44'/1'/0'"? Ledger will give me the error `"Ledger device: Internal error, please report (0x6f15)"`.